File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        int mid = 0;
        while(left <= right){
             mid = left + (right - left)/ 2;

           if(target <= nums[mid]){
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }
        return left;
        
    }
}

File: 0035-search-insert-position/0035-search-insert-position.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut left = 0;
        let mut right = nums.len();
        while left < right{
            let mid = left +(right - left) / 2;
            if nums[mid] == target{
                return mid as i32;
            }
            if nums[mid] >= target{ //before the mid
                right = mid;
            }else{
                left = mid + 1;
            }
        }
        left as i32
    }
}